quickwit/quickwit-ingest/src/ingest_v2/ingester.rs

```diff
@@ -835,6 +829,78 @@ impl IngesterService for Ingester {
     }
 }
 
+impl IngesterState {
+    /// Truncates the shard identified by `queue_id` up to `truncate_position_inclusive` only if the
+    /// current truncation position of the shard is smaller.
+    async fn truncate_shard(&mut self, queue_id: &QueueId, truncate_position_inclusive: Position) {
+        if let Some(truncate_offset_inclusive) = truncate_position_inclusive.as_u64() {
```

Comment:
```suggestion
        let Some(truncate_offset_inclusive) = truncate_position_inclusive.as_u64() {
             return;
        }; 
```

early-return style is nice to avoid that kind of nesting.

Reply:
Replaced with a bunch of early returns. This snippet is a good contender for if let chains when stabilized.